imx/assets.go

```diff
@@ -0,0 +1,54 @@
+package imx
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/immutable/imx-core-sdk-golang/imx/api"
+)
+
+/*
+GetAsset Get details of an asset
+
+@param ctx context.Context - for cancellation, deadlines, tracing, etc or context.Background().
+@param tokenAddress Address of the ERC721 contract
+@param tokenId Either ERC721 token ID or internal IMX ID
+@param includeFees optional param.
+@return Asset
+*/
+func (c *Client) GetAsset(ctx context.Context, tokenAddress, tokenId string, includeFees *bool) (*api.Asset, error) {
+	apiGetAssetRequest := c.assetsApi.GetAsset(ctx, tokenAddress, tokenId)
+	if includeFees != nil {
+		apiGetAssetRequest.IncludeFees(*includeFees)
+	}
+
+	response, httpResponse, err := apiGetAssetRequest.Execute()
+	if err != nil {
```

Comment:
Instead of handling the errors we might want to return them unaltered. 
The way it is now, the user will not be able to parse the http status code etc easily.
We don't have this problem in typescript sdk because the response there is a `Promise<AxiosResponse<Type...`

Reply:
Fair point. This is a pattern followed all across SDK, this will mean every method will need to be updated. If we want to do this, it will be a seperate task.

Reply:
I disagree that this is out of scope of this ticket as the issue is caused by the changes in this PR.

[DX-1162] limit public api functions accessible by user:

Currently user is able to directly access methods e.g. get x and handle errors differently. A 404 would probably be handled differently from a 500, or 40x. 
The wrapped errors in this PR makes it harder to do this.

Reply:
Ideally, we align with the public-api's custom error pattern: introduce an `SDKError` (or IMXError, the naming is TBC) type and have it implement the `error` interface. 

```go
func (app *Instance) getAsset(w http.ResponseWriter, / *...args */) (*response.APIResponse, *response.APIError)
```

In addition, it's a good idea also to follow the behavioral error pattern here:

https://github.com/immutable/imx-engine/tree/main/common/errors

This brings great ergonomics that developers would love (and eventually ask for).

If we don't do it this time, we should aim to address this in the next round.

Reply:
In typescript sdk, the team is looking to return instead the APIError object that comes from public api openapi spec. 
<img width="555" alt="image" src="https://user-images.githubusercontent.com/5828814/191133459-e357e238-4c0b-4511-8490-94d7b782724c.png">

Reply:
Will update spec and look at this as a seperate task.